### PR TITLE
add sharing management

### PR DESCRIPTION
**Pull request by WeidiDeng**
**Description**
Added sharing management. Normal user can only see own shares, admin sees all.
Fixes [issuse](https://github.com/filebrowser/filebrowser/issues/1000)

:rotating_light: Before submitting your PR, please read [community](https://github.com/filebrowser/community), and indicate which issues (in any of the repos) are either fixed or closed by this PR. See [GitHub Help: Closing issues using keywords](https://help.github.com/articles/closing-issues-via-commit-messages/).

- [ ] DO make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [ ] DO make sure you are making a pull request against the **master branch** (left side). Also you should start *your branch* off *our master*.
- [ ] DO make sure that File Browser can be successfully built. See [builds](https://github.com/filebrowser/community/blob/master/builds.md) and [development](https://github.com/filebrowser/community/blob/master/development.md).
- [ ] DO make sure that related issues are opened in other repositories. I.e., the frontend, caddy plugins or the web page need to be updated accordingly.
- [ ] AVOID breaking the continuous integration build.

**Further comments**
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did, what alternatives you considered, etc.

:heart: Thank you!

### CODE REVIEW

File: frontend/src/views/settings/Shares.vue [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        for (let link of links) link.username = userMap.has(link.userID) ? userMap.get(link.userID) : ''
+      }
+      this.links = links
+    } catch (e) {
+      this.$showError(e)
+    }
+  },
+  mounted() {
+    this.clip = new Clipboard('.copy-clipboard')
+    this.clip.on('success', () => {
```
Comment (ramiresviana) [inline frontend/src/views/settings/Shares.vue:73]: This event listener should be removed when the component is destroyed, similar to the share prompt.

**Review by ramiresviana — CHANGES_REQUESTED** [review]

File: http/http.go [hunk 2]
```diff
@@ -51,6 +51,7 @@ func NewHandler(imgSvc ImgService, fileCache FileCache, store *storage.Storage,
 	api.PathPrefix("/resources").Handler(monkey(resourcePostPutHandler, "/api/resources")).Methods("PUT")
 	api.PathPrefix("/resources").Handler(monkey(resourcePatchHandler, "/api/resources")).Methods("PATCH")
 
+	api.Path("/shares").Handler(monkey(shareListHandler, "")).Methods("GET")
```
Comment (o1egl) [inline http/http.go]:
```suggestion
	api.Path("/shares").Handler(monkey(shareListHandler, "/api/shares")).Methods("GET")
```

File: share/storage.go [hunk 3]
```diff
@@ -8,6 +8,7 @@ import (
 
 // StorageBackend is the interface to implement for a share storage.
 type StorageBackend interface {
+	List(id uint) ([]*Link, error)
```
Comment (o1egl) [inline share/storage.go]: Let's rename it to `FindByUserID`

File: http/share.go [hunk 4]
```diff
@@ -23,6 +24,30 @@ func withPermShare(fn handleFunc) handleFunc {
 	})
 }
 
+var shareListHandler = withPermShare(func(w http.ResponseWriter, r *http.Request, d *data) (int, error) {
+	id := d.user.ID
+	if d.user.Perm.Admin {
+		id = 0
+	}
+	s, err := d.store.Share.List(id)
```
Comment (o1egl) [inline http/share.go]: Instead of defining magic constants let's add `All()` method to the store for admin users

**Review by o1egl — CHANGES_REQUESTED** [review]

**Review by o1egl — APPROVED** [review]

**Pull request merged**